examples/default/main.tf

```diff
@@ -19,19 +19,17 @@ data "ibm_resource_group" "existing_resource_group" {
 #############################################################################
 
 resource "ibm_resource_instance" "cos_instance" {
-  count = var.enable_vpc_flow_logs ? 1 : 0
-
-  name              = "${var.prefix}-vpc-logs-cos"
+  count             = var.enable_vpc_flow_logs ? 1 : 0
+  name              = var.prefix != null ? "${var.prefix}-vpc-logs-cos" : "vpc-logs-cos"
   resource_group_id = var.resource_group != null ? data.ibm_resource_group.existing_resource_group[0].id : ibm_resource_group.resource_group[0].id
   service           = "cloud-object-storage"
   plan              = var.cos_plan
   location          = var.cos_location
 }
 
 resource "ibm_cos_bucket" "cos_bucket" {
-  count = var.enable_vpc_flow_logs ? 1 : 0
-
-  bucket_name          = "${var.prefix}-vpc-logs-cos-bucket"
+  count                = var.enable_vpc_flow_logs ? 1 : 0
+  bucket_name          = var.prefix != null ? "${var.prefix}-vpc-logs-cos-bucket" : "vpc-logs-cos-bucket"
```

Comment:
I would not add this support to the example. The example should always use a prefix (so we avoid name clashes during test runs in our account)

Reply:
@Khuzaima05 please remove this change. This does not have to support passing prefix as null. Infact prefix is required in the **example**